aiida/engine/daemon/execmanager.py

```diff
@@ -139,8 +139,12 @@ def upload_calculation(node, transport, calc_info, folder, inputs=None, dry_run=
     for code in input_codes:
         if code.is_local():
             # Note: this will possibly overwrite files
-            for f in code.get_folder_list():
-                transport.put(code.get_abs_path(f), f)
+            for f in code.list_object_names():
+                with NamedTemporaryFile(mode='wb+') as handle:
+                    handle.write(code.get_object_content(f, mode='rb'))
+                    handle.flush()
+                    handle.seek(0)
```

Comment:
Is the `seek` really necessary here? Doesn't `transport.put` open it itself anyway? If it's necessary, might warrant a comment as to why.

Reply:
You are right, it is not necessary, I will remove it. The code is a copy of a block a few lines lower. There I already put a comment that it should be improved using `node.open` once `transport.put` supports filelike objects to be passed, which is issue #2579 . I decided not to copy the comment again.